src/main/java/org/craftercms/studio/impl/v2/service/proxy/internal/ProxyServiceInternalImpl.java

```diff
@@ -69,49 +74,104 @@ public ResponseEntity<Object> getSiteLogEvents(final String body,
+        String proxyPath = getProxyPath(request);
+        List<String> engineProtectedUrls = getEngineProtectedUrls();
+        boolean managementTokenRequired = engineProtectedUrls.contains(proxyPath);
         URI uri = new URI(getAuthoringUrl(siteId));
         UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromUri(uri)
-                .path(proxiedUrl)
+                .path(proxyPath)
                 .query(request.getQueryString())
                 .replaceQueryParam("site", siteId);
         if (managementTokenRequired) {
             uriComponentsBuilder = uriComponentsBuilder.queryParam("token", getEngineManagementTokenValue());
         }
         uri = uriComponentsBuilder.build(true).toUri();
 
-        // Copy all headers
+        return uri;
+    }
+
+    /**
+     * Returns headers which should be passed to the proxy server (internal engine server)
+     * @param request the current request from Studio
+     * @return headers sending to proxy server
+     */
+    private HttpHeaders getProxyRequestHeaders(HttpServletRequest request) {
         HttpHeaders headers = new HttpHeaders();
         Enumeration<String> headerNames = request.getHeaderNames();
+        // Add all headers except an ignored list and the cookie header
         while (headerNames.hasMoreElements()) {
             String headerName = headerNames.nextElement();
-            headers.set(headerName, request.getHeader(headerName));
+            if (!IGNORE_REQUEST_HEADERS.contains(headerName.toLowerCase()) && !headerName.equalsIgnoreCase(HttpHeaders.COOKIE)) {
+                headers.set(headerName, request.getHeader(headerName));
+            }
         }
 
-        // Execute proxied request and return response
-        HttpEntity<Object> httpEntity = new HttpEntity<>(body, headers);
-        try {
-            ResponseEntity res = restTemplate.exchange(uri, HttpMethod.valueOf(request.getMethod()), httpEntity, Object.class);
-            HttpHeaders resHeaders = new HttpHeaders();
-            // https://www.rfc-editor.org/rfc/rfc9112#name-transfer-encoding
-            // A sender MUST NOT apply the chunked transfer coding more than once to a message body
-            res.getHeaders().forEach((key, value) -> {
-                if (!key.equals(HttpHeaders.TRANSFER_ENCODING)) {
-                    resHeaders.addAll(key, value);
+        // rebuild cookie headers to remove ignored list of cookies
+        String proxyCookieHeader = "";
+        Cookie[] cookies = request.getCookies();
+        for (Cookie cookie : cookies) {
+            if (!IGNORE_REQUEST_COOKIES.contains(cookie.getName().toLowerCase())) {
+                if (StringUtils.isEmpty(proxyCookieHeader)) {
+                    proxyCookieHeader = format("%s=%s", cookie.getName(), cookie.getValue());
+                } else {
+                    proxyCookieHeader += format("; %s=%s", cookie.getName(), cookie.getValue());
```

Comment:
There might be many cookies, let's use StringBuilder instead of `+=`

Reply:
Updated to use `StringBuilder`. Thanks!

Reply:
This function has been moved to commons: https://github.com/craftercms/commons/pull/327